internal/authz/authz.go

```diff
@@ -195,6 +276,49 @@ func (a *Authorizer) individualAuthorize(ctx context.Context, clientID, relation
 	return nil
 }
 
+// moduleAuthorize checks if the user has access to each of the modules, and exits if an error is encountered.
+func (a *Authorizer) moduleAuthorize(ctx context.Context, clientID, relation, storeID string, modules []string) error {
```

Comment:
Can I please get a review of this channel code? Basically I am concerned that there might be a case I'm not handling where there is an error in the modules, but the channel closes before the error is added to the errorChannel. @adriantam

Reply:
The problem is that there is no guarantee that line 308-312 are executed.  hence, there may be cases where the channels are closed.

Reply:
Refactored as per @senojj suggestion